### ISSUE TITLE
form serialization should be optional

### ISSUE THREAD

**Issue by leastbad** [hopsoft/stimulus_reflex#422]
Forms should not automatically serialize to the `params` collection.

This behaviour should be opt-in, either globally or on a per-Reflex basis.

Switching from using `form` elements to something like `input-group` might be more logical. Or perhaps we introduce the `data-reflex-serialize` attribute and allow users to put it on any element they like. The functionality could behave exactly as it does today, except looking for that attribute instead of a `form` ancestor.

This is my current top suggestion for how to go about things:

```html
<div data-reflex-serialize>
  <input type="text" data-reflex="change->Foo#bar" name="foo[name]">
  <input type="text" data-reflex="change->Foo#baz" name="foo[age]">
</div>
```

In other words, it behaves **exactly** like today's form serialization does, except that instead of implicitly assuming serialize is desired because it's in a `form` element, everything is now explicit and not tied to the `form` element. Which means we could stop referring to this as "form serialization".

People would of course be still welcome to use `<form data-reflex-serialize>` but it wouldn't have any meaningful impact on the behaviour of the functionality.

**Comment by leastbad** [comment]
Form serialization will become opt-in in v4, a flip of default behaviour and breaking change from v3.x.

v3.5 will introduce Deprecation Warnings in development mode, and `data-reflex-serialize-form` is already on the list.

**Comment by iraszl** [comment]
What do you need to do to make this warning go away?
